stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -332,6 +333,31 @@ static int _addImageProtocolConformances(struct dl_phdr_info *info,
   dlclose(handle);
   return 0;
 }
+#elif defined(__CYGWIN__)
+static int _addImageProtocolConformances(struct dl_phdr_info *info,
```

Comment:
That would be great! [Android armv7](https://github.com/SwiftAndroid/swift) also has this sort of duplication:

- [`stdlib/public/runtime/MetadataLookup.cpp`](https://github.com/SwiftAndroid/swift/commit/32391abf733831131939c82d1d5d7eefd8d3743d)
- [`stdlib/public/runtime/ProtocolConformance.cpp`](https://github.com/SwiftAndroid/swift/blob/52fbae5fb9cd1df1ea6649f8382d776b850a7447/stdlib/public/runtime/ProtocolConformance.cpp#L310-L369)

It would be awesome if this could be factored so that it's easier for all ports. :+1: